lib/EnsEMBL/REST/Model/Eqtl.pm

```diff
@@ -0,0 +1,152 @@
+
+  # fails in registry if species is wrong/not available in Ensembl
+  my $user_species    = $constraints->{species};
+  my $ensembl_species = $self->{'registry'}->get_alias($user_species);
+
+  if(! $ensembl_species) {
+    Catalyst::Exception->throw("Species '$user_species' not recognised. Check your URL");
+  }
+
+  my $eqtl_a = $self->{'registry'}->get_eqtl_adaptor($ensembl_species);
+
+  $self->_validate_stable_id ($eqtl_a, $constraints->{stable_id});
+  $self->_validate_tissuei   ($eqtl_a, $constraints->{tissue});
+  $self->_validate_statistic ($eqtl_a, $constraints->{statistic});
+
+  my $results = $eqtl_a->fetch( {
+        gene          => $constraints->{stable_id},
+        tissue        => $constraints->{tissue},
+        snp           => $constraints->{variant_name},
+        statistic     => $constraints->{statistic},
+      });
+
+  return $results;
+}
+
+=head2 _validate_stable_id
+
+  Arg [1]    : Bio::EnsEMBL::HDF5::EQTLAdaptor
+  Arg [2]    : $stable_id
+  Example    : $self->_validate_stable_id($eqtl_a, $constraints->{stable_id});
+  Description: Validates if stable_id exists in Database.
+  Returntype : None
+  Exceptions : Throws if stable_id is not present in Database
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub _validate_stable_id {
+  my ($self, $eqtl_a, $stable_id) = @_;
+  # Check for EnsemblIDs with more/less than 11 digits
+  if (defined $stable_id){
+    if( ($stable_id =~ /ENS[A-Z]+[0-9]{11}/)  and (! exists $eqtl_a->{gene_ids}->{$stable_id}) ) {
+      Catalyst::Exception->throw("Stable ID '$stable_id' not present in the current Ensembl release");
+    }
+  }
+}
+
+=head2 _validate_tissue
+
+  Arg [1]    : Bio::EnsEMBL::HDF5::EQTLAdaptor
+  Arg [2]    : $tissue
+  Example    : $self->_validate_tissue($eqtl_a, $constraints->{tissue});
+  Description: Validates if tissue exists in Database.
+  Returntype : None
+  Exceptions : Throws if tissue is not present in Database
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub _validate_tissue {
+  my ($self, $eqtl_a, $tissue) = @_;
+  if (defined $tissue){
+    if(! exists $eqtl_a->{tissue_ids}->{$tissue}) {
+      my $tissues = join(", ", map { $_ } sort keys %{$eqtl_a->{tissue_ids}});
+      Catalyst::Exception->throw("Tissue '$tissue' not recognised. Available tissues: $tissues");
+    }
+  }
+}
+
+=head2 _validate_statistic
+
+  Arg [1]    : Bio::EnsEMBL::HDF5::EQTLAdaptor
+  Arg [2]    : $statistic name
+  Example    : $self->_validate_statistic($eqtl_a, $constraints->{statistic});
+  Description: Validates if statistic exists in Database.
+  Returntype : None
+  Exceptions : Throws if statistic is not present in Database
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub _validate_statistic {
+  my ($self, $eqtl_a, $statistic) = @_;
+
+  if (defined $statistic){
+    if(! exists $eqtl_a->{statistic_ids}->{$statistic}) {
+      my $statistics = join(", ", map { $_ } sort keys %{$eqtl_a->{statistic_ids}});
```

Comment:
As discussed elsewhere, you're doing work for every request, that might be preloaded and cached. It's fine for your current size of list, but has the potential to grow. See also _validate_tissue